integration-tests/test_obfuscation.py

```diff
@@ -1,71 +1,168 @@
-        cmd = ["tar", "-vxzf", archive_location, "-C", tmp_path]
-        subprocess.check_call(cmd)
-
-        # the redaction data should not locate in the collected files
-        archive_extract = os.path.basename(archive_location).split(".")[0]
-        path_to_check = tmp_path / archive_extract
-        cmd = ["grep", "-rnw", ".", "-e", hostname_to_test]
-        # should not find the obfuscate pattern, the return code should not zero.
-        ret = subprocess.run(cmd, cwd=path_to_check)
-        assert ret.returncode == 1
+        # Add a line about password in a tested file
+        password_string = "my-super-secret-password"
+        with open(password_file, "a") as file:
+            file.write(f"\npassword: {password_string}")
+
+        # Make sure obfuscate=False
+        insights_client.config.obfuscate = False
+        insights_client.config.save()
+
+        # the password value is not visible although obfuscate=False
+        assert not check_obfuscated_info(insights_client, tmp_path, password_string)
+
+        # the password  value is not visible after Obfuscating IP
+        insights_client.config.obfuscate = True
+        insights_client.config.save()
+        assert not check_obfuscated_info(insights_client, tmp_path, password_string)
     finally:
-        # Recover the previous hostname for the system
-        set_hostname(previous_hostname)
-        # Remove all data redaction
-        os.remove(CONTENT_REDACTION_FILE)
+        with open(password_file, "w+") as f:
+            f.write(original_content)
+
+
+@pytest.mark.parametrize(
+    "package_info_file",
+    [
+        "data/insights_commands/rpm_-qa_--qf",
+        "data/insights_commands/yum_-C_--noplugins_list_available",
+        "data/insights_commands/yum_updates_list",
+        "data/etc/dnf/modules.d/",
+    ],
+)
+def test_no_obfuscation_on_package_version(
+    insights_client, tmp_path, package_info_file
+):
+    """Test exclusion of certain Specs from IP address obfuscation:
+    Examine the files produced by these Specs:
+            • insights.specs.Specs.installed_rpms
+            • insights.specs.Specs.dnf_modules
+            • insights.specs.Specs.yum_list_available
+            • insights.specs.Specs.yum_updates
+    Verify that version strings consisting of four parts
+         like shadow-utils-4.1.5.1-24.el7.x86_64 are not obfuscated
+         as if they were IPv4 addresses. An obfuscated string would
+         look like this: shadow-utils-10.230.230.21-24.el7.x86_64
+    Related Bug: https://issues.redhat.com/browse/ESSNTL-444
+    """
+    # Obfuscate IP
+    insights_client.config.obfuscate = True
+    insights_client.config.save()
+
+    # Check if the package version is obfuscated by IP obfuscation
+    archive_name = "test_archive_" + str(uuid.uuid4()) + ".tar.gz"
+    archive_location = tmp_path / archive_name
+    insights_client.run(
+        "--register", "--no-upload", "--output-file=%s" % archive_location
```

Comment:
no need for --no-upload, because --output-file runs in offline mode only.

Reply:
Done